torchtitan/metrics.py

```diff
@@ -126,35 +162,69 @@ def _get_metrics_rank(parallel_dims: ParallelDims) -> int:
         metrics_log_rank = (world_size // pp_size) * (pp_size - 1)
     else:
         metrics_log_rank = 0
-
     return metrics_log_rank
 
 
 def build_metric_logger(
     job_config: JobConfig, parallel_dims: ParallelDims, tag: Optional[str] = None
-):
+) -> Union[DummyLogger, TensorBoardLogger, WandBLogger]:
     """
-    parallel_dims is used to determine the rank to log metrics from if 'tb_config.rank_0_only=True'.
-    In that case, `_get_metrics_rank` will be used to calculate which rank acts as 'rank 0'. This is
-    intended to allow logging from the 0th rank within the last pipeline stage group, in case pipeline
-    parallelism is enabled, without forcing logging from all ranks to capture loss information.
+    Build an appropriate metric logger based on configuration.
     """
+    metrics_config = job_config.metrics
+
+    # Log initial config state
+    logger.info(
+        f"Building logger with config: wandb={metrics_config.enable_wandb}, "
+        f"tensorboard={metrics_config.enable_tensorboard}"
+    )
```

Comment:
It seems the `logger.info` from loggers is a bit verbose.
I suggest we keep only two by doing: first being the one here, second inside `__init__` of loggers.
Other loggings, if necessary, can go to `logger.debug` (or `logger.error` for the import error one). What do you think?

Reply:
Oh this particular statement I used to debug a confusing a bug around the default behavior of flags, it was useful so I might just make this a debug statement